mapbox_tilesets/scripts/cli.py

```diff
@@ -582,18 +575,13 @@ def _upload_source(
                 f"Token {mapbox_token} does not contain a username"
             )
 
-    with tempfile.TemporaryFile() as file:
-        for index, feature in enumerate(features):
-            if not no_validation:
-                utils.validate_geojson(index, feature)
-
-            file.write(
-                (json.dumps(feature, separators=(",", ":")) + "\n").encode("utf-8")
-            )
-
-        file.seek(0)
-        m = MultipartEncoder(fields={"file": ("file", file)})
+    if len(inputs) > 10:
+        raise errors.TilesetsError("Maximum 10 files can be uploaded at once.")
 
+    for item in inputs:
```

Comment:
The current `upload-source` command works slightly differently with multiple GeoJSON files than what's written here.  Even though it accepts multiple input GeoJSON files, it actually concatenates all the features from all files into a single `ldgeojson` file, then uploads that combined file as a single source.

As written here, these changes will actually upload up to 10 individual sources to the same Source ID (I'm not sure if that fails or just overwrites the source--would have to check api-tilesets).

Let's discuss options tomorrow morning - happy to pair on this and think through options!